brave/src/main/java/com/linecorp/armeria/client/brave/BraveClient.java

```diff
@@ -68,98 +66,116 @@
 
     /**
      * Creates a new tracing {@link Client} decorator using the specified {@link Tracing} instance
-     * and remote service name.
+     * and the remote service name.
      */
     public static Function<Client<HttpRequest, HttpResponse>, BraveClient> newDecorator(
             Tracing tracing, @Nullable String remoteServiceName) {
+        HttpTracing httpTracing = HttpTracing.newBuilder(tracing)
+                                             .clientParser(new ArmeriaHttpClientParser())
+                                             .build();
+        if (remoteServiceName != null) {
+            httpTracing = httpTracing.clientOf(remoteServiceName);
+        }
+        return newDecorator(httpTracing);
+    }
+
+    /**
+     * Creates a new tracing {@link Client} decorator using the specified {@link HttpTracing} instance.
+     */
+    public static Function<Client<HttpRequest, HttpResponse>, BraveClient> newDecorator(
+            HttpTracing httpTracing) {
         try {
-            ensureScopeUsesRequestContext(tracing);
+            ensureScopeUsesRequestContext(httpTracing.tracing());
```

Comment:
Should we have `httpTracing.toBuilder().clientParser(new ArmeriaHttpClientParser())`?

Reply:
that would overwrite what the user supplied so I think no

Reply:
actually the line mention isnt using http stuff anyway so it shouldn't matter I guess

Reply:
I guess we'd need to check if the user passed anything first. I didn't mean to specifically point to this line - but if the user didn't set client parser correctly, I think the tracing wouldn't work so we should probably give it a default when not specified.

Reply:
Default implementation of https tracing will use http client/server parser directly (only has few taga I think)

Reply:
yeah if there was a problem due to no data in defaults it would affect everyone. Right now the default parsing is conservative but understood. OTOH, if someone is not passing an HttpTracing (ex they pass Tracing) then it is good to use the armeria parsers because that gives same behavior as before.